allow blocking on the listen, and typo fix

dagrh opened the pull request:
I'm trying to write a vhost daemon which will wait for the connection from qemu; so I want to be able to wait on the listen.

dagrh committed:
connection: Allow changing the blocking setting

Some users want to wait for the socket to be opened,
so provide a set_nonblocking wrapper.

Signed-off-by: Dr. David Alan Gilbert <dgilbert@redhat.com>

dagrh committed:
Typo: underline->underlying

copied in a few places.

Signed-off-by: Dr. David Alan Gilbert <dgilbert@redhat.com>

jiangliu reviewed (commented):
Could you please help to also remove calling to set_nonblocking() in Listener::new()? Once we implement Listener::set_nonblocking(), it would be better to keep the socket in blocking mode by default.
Thanks!

dagrh committed:
test: Fix VhostUserProtocolFeatures::all comparison

Signed-off-by: Dr. David Alan Gilbert <dgilbert@redhat.com>

dagrh committed:
connection/listener: Make blocking by default

Make the listener blocking by default, that's the more common
use case for the daemon.
Fix up all the tests so that they still unblock (since they're
simple single threaded tests).

Signed-off-by: Dr. David Alan Gilbert <dgilbert@redhat.com>

dagrh commented:
Hi,
  I've just pushed an updated version that includes a test fix and then flips the default blocking around on Listener::new() - is that what you were hoping for?
